torch/csrc/utils/tensor_new.cpp

```diff
@@ -121,8 +121,7 @@ ScalarType infer_scalar_type(PyObject *obj) {
     return ScalarType::Long;
   }
   if (PyBool_Check(obj)) {
-    // TODO: infer Bool when we have Bool ScalarType
-    return ScalarType::Byte;
+    return ScalarType::Bool;
```

Comment:
all of the non-test changes besides this look correct and can go in (because they don't break backcompat), right?

Reply:
this and jit changes